components/EntryForm.js

```diff
@@ -1,10 +1,43 @@
 import styled from "styled-components";
 import WrappedDatePicker from "./WrappedDatePicker";
+import { calculator } from "@/library/calculator";
 import { useState } from "react";
 
 export default function EntryForm() {
   const [startDate, setStartDate] = useState(new Date());
   const [transport, setTransport] = useState("Select a transport");
+  const [fuel, setFuel] = useState("");
+  const [km, setKm] = useState(0);
+  const [result, setResult] = useState(0);
+
+  function handleCalculateCo2(transport, km, fuel) {
+    if ((transport === "car") && (fuel === "petrol")) {
+      return calculator[0].petrol(km);
+    } else if 
+      ((transport === "car") && (fuel === "diesel")) {
+        return calculator[1].diesel(km);
+    } else if 
+      ((transport === "car") && (fuel === "hybrid")) {
+        return calculator[2].hybrid(km);
+    } else if 
+      ((transport === "car") && (fuel === "electric-strommix")) {
+        return calculator[3].electricStrommix(km);
+    } else if 
+      ((transport === "car") && (fuel === "electric-renewable")) {
+        return calculator[4].electricEco(km);
+    } else if 
+      (transport === "plane") {
+        return calculator[5].plane(km);
+    } else if 
+      (transport === "train") {
+        return calculator[6].train(km);
+    } else if 
+      (transport === "bicycle") {
+        return calculator[7].bicycle(km);
+    } else {
+      console.log("Please select a transport");
+    }
+  }
```

Comment:
I would normalise the calculators as a plain object:

```js
export const calculator = {
  petrolCar: (km) => {
    const petrolEmissionInKg = 0.189;
    return Math.round(petrolEmissionInKg * km * 100) / 100;
  },
  // average value of 7l/100km -- average emission of 2.65kg/l
  dieselCar: (km) => {
    const dieselEmissionInKg = 0.1855;
    return Math.round(dieselEmissionInKg * km * 100) / 100;
  },
  // average value of 4l/100km -- average emission of 2.37kg/l
  hybridCar: (km) => {
    const hybridEmissionInKg = 0.0948;
    return Math.round(hybridEmissionInKg * km * 100) / 100;
  },
  // average value of 18kWh/100km -- average emission of 0,434kg/kWh
  electricStrommixCar: (km) => {
    const strommixEmissionInKg = 0.07812;
    return Math.round(strommixEmissionInKg * km * 100) / 100;
  },
  electricEcoCar: (km) => {
    const ecoEmissionInKg = 0.0;
    return ecoEmissionInKg * km;
  },
  plane: (km) => {
    const planeEmissionInKg = 0.23;
    return Math.round(planeEmissionInKg * km * 100) / 100;
  },
  // average value for long-distance traffic
  train: (km) => {
    const trainEmissionInKg = 0.032;
    return Math.round(trainEmissionInKg * km * 100) / 100;
  },
  bicycle: (km) => {
    const bicycleEmissionInKg = 0.0;
    return bicycleEmissionInKg * km;
  },
};
```

Then you'll need just a single `<select>`, with all the options:

```html
<select>
  <option value="petrolCar">Car - Petrol</option>
  <option value="dieselCar">Car - Diesel</option>
  ...
  <option value="plane">Plane</option>
  ...
</select>
```

(you can still generate it via `map()` - good!)

At the end in `handleCalculateCo2` you drop the `fuel` parameter (and from the state as well):

```js
function handleCalculateCo2(transport, km) {
  return calculator[transport](km);
}
```

Reply:
Hi Diego, thanks a lot for your feedback! I will change my code as you recommended.